aws/resource_aws_iam_role.go

```diff
@@ -239,6 +264,21 @@ func resourceAwsIamRoleUpdate(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
+	if d.HasChange("max_session_duration") {
+		roleMaxDurationInput := &iam.UpdateRoleInput{
+			RoleName:           aws.String(d.Id()),
+			MaxSessionDuration: aws.Int64(d.Get("assume_role_policy").(int64)),
+		}
+		_, err := iamconn.UpdateRole(roleMaxDurationInput)
+		if err != nil {
+			if iamerr, ok := err.(awserr.Error); ok && iamerr.Code() == "NoSuchEntity" {
```

Comment:
Two notes here:
* This can be simplified and use the SDK provided constant with `if isAWSErr(err, iam.ErrCodeNoSuchEntityException, "") {`
* The error message needs to be updated